### ISSUE TITLE
关于SimGCL与XSimGCL所使用的 InfoNCE 的问题

### ISSUE THREAD

**Issue by Zeng-CG** [Coder-Yu/SELFRec#48]
作者您好，有幸拜读了您的两篇杰出工作SimGCL与XSimGCL，特别是XSimGCL它是那么简单优雅并且有效。

我在关注两个模型所使用的InfoNCE函数时发现它只关注正样本之间的相似度，而对负样本之间的相似度不关心，请问这是出于什么目的呢？（有可能这个问题比较简单，如能答复不胜感激）

```
def InfoNCE(view1, view2, temperature: float, b_cos: bool = True):
    if b_cos:
        view1, view2 = F.normalize(view1, dim=1), F.normalize(view2, dim=1)
    pos_score = (view1 @ view2.T) / temperature
    score = torch.diag(F.log_softmax(pos_score, dim=1))
    return -score.mean()
```

也是因为与XsimGCL论文中损失函数不一致让我感到困惑

**Comment by Coder-Yu** [comment]
`也是因为与XsimGCL论文中损失函数不一致让我感到困惑`
不一致在？

`它只关注正样本之间的相似度，而对负样本之间的相似度不关心`
关注负样本的相似度没有意义，除非考虑false negatives。Infonce的目的是最大化同一个节点的不同视图的互信息。

**Comment by Zeng-CG** [comment]
谢谢作者的回复，我应该是没有注意这行代码中F.log_softmax的作用，多有打扰
`score = torch.diag(F.log_softmax(pos_score, dim=1))`
